crates/metadata-server/src/raft/server/member.rs

```diff
@@ -290,16 +359,38 @@ impl Member {
             .wait_for_version(MetadataKind::NodesConfiguration, nodes_config.version())
             .await?;
 
-        // todo if I am the leader, then tell others to immediately start campaigning to avoid the leader election timeout
-        Ok(Standby::new(
-            storage,
-            self.connection_manager,
-            self.request_rx,
-            self.join_cluster_rx,
-            self.metadata_writer,
-            self.status_tx,
-            self.command_rx,
-        ))
+        Ok(())
+    }
+
+    /// Shuts the member down by failing all pending requests and transferring leadership if it is the
+    /// leader.
+    pub async fn shutdown(&mut self) -> Result<(), Error> {
+        let nodes_config = Metadata::with_current(|m| m.nodes_config_ref());
+        self.fail_pending_requests(&nodes_config);
+
+        if self.is_leader {
+            debug!(
+                "Shutting down as leading member. Trying to transfer leadership to an alive node."
+            );
+
+            let cluster_state = TaskCenter::with_current(|h| h.cluster_state().clone());
+            if let Some(transferee) = self.configuration.members.keys().find(|&member| {
+                member != &my_node_id().as_plain() && cluster_state.is_alive(NodeId::from(member))
+            }) {
+                info!("Transferring metadata cluster leadership to {transferee}.");
+                let transferee = to_raft_id(*transferee);
+                self.raw_node.transfer_leader(transferee);
+
+                // run one round of ready processing to send out leader transfer messages
+                self.on_ready(&nodes_config).await?;
+            } else {
+                debug!(
+                    "Failed transferring metadata cluster leadership because no member is alive."
```

Comment:
can we transfer even if only one member is alive? what happens if there is not quorum when we try to transfer?